docs/Shaders-and-Smoothing.md

```diff
@@ -42,42 +46,55 @@ Shaders are small programs that a dedicated graphics chip (GPU) runs to alter th
 
 The second two presets are `crt-pi-vertical.glslp` and `crt-pi-curvature-vertical.glslp`, described in [Vertical games](#vertical-games)
 
-## Installation
+#### zfast
+
+[@ghogan42](https://retropie.org.uk/forum/user/ghogan42) has built shader **zfast**, which seems to perform even better. Five presets are included in RetroPie:
+
+1. `zfast_crt_standard.glslp`
+![](https://i.imgur.com/KnUBAMu.jpg)
+
+2. `zfast_crt_curve.glslp`
+![](https://i.imgur.com/9cnVeTC.jpg)
+
+3. `zfast_lcd_standard.glslp`
+![](https://i.imgur.com/WOQXb1z.jpg)
 
-**Note:** Shaders can only be used in Libretro emulators. Most default emulators in RetroPie are Libretro emulators, but you can confirm by looking for the lr- prefix in the emulator name via the [Runcommand](Runcommand).
+The final 2 presets are `zfast_crt_standard_vertical.glslp` and `zfast_crt_curve_vertical.glslp` for vertical games, as with [crt-pi](#crt-pi).
 
-You can select either preset via the [Configuration Editor](Configuration-Editor). Choose `Configure basic libretro emulator options`, then choose a specific emulator or `Configure default options for all libretro emulators` to make the change system-wide. Then, set `Video Shader Enable` to `true` and then set `Video Shader File` to the desired shader.
+#### Installation
 
-### Scaling artifacts (particularly in Vertical games)
+**Note:** These shaders can only be used in Libretro emulators. Most default emulators in RetroPie are Libretro emulators, but you can confirm by looking for the `lr-` prefix in the emulator name via the [Runcommand](Runcommand).
 
-Some arcade games were run on a vertical CRT; literally a regular TV flipped over on its side. Whilst libretro shaders  automatically rotate for this, the shadow mask effect does not, causing an unsightly 'rainbow' effect. Rather than add logic to the shader (making it slower), a special version for vertical games is available:
+You can select either preset via the [Configuration Editor](Configuration-Editor). Choose **Configure basic libretro emulator options**, then choose a specific emulator or **Configure default options for all libretro emulators** to make the change system-wide. Then, set `Video Shader Enable` to `true` and then set `Video Shader File` to the desired shader.
+
+You can can save per-game (or per-core) within a game via the **RGUI** with player 1's [Hotkey](Controller-Configuration#Hotkey) combination **Hotkey+X**, then **Quick Menu** > **Shaders**:
+
+![](https://i.imgur.com/EOCVSbT.png?1)
```

Comment:
The screenshot could be updated, there's a single 'Save' menu now, with multiple options on the next menu: https://imgur.com/a/gEK4IWF